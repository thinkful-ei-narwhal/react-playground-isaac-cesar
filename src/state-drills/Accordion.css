li button {
	padding: 0.5rem;
	background-color: blue;
	color: #fff;
	font-size: 1rem;
	margin: 0.5rem;
}

li p {
	padding: 2rem;
	background-color: bisque;
}
