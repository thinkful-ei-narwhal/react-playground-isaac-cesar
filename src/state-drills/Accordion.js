import React from 'react';

class Accordion extends React.Component {
	state = {
        content: '',
        indexClick: null
	};

	handleDisplay() {
		return this.props.sections.map((e, index) => {
			return (
				<li key={index}>
					<button
						onClick={() => {
							return this.setState({ 
                                content: e.content,
                                indexClick: index
                            });
						}}
					>
						{e.title}
					</button>
                    {(this.state.indexClick=== index)&&<p>{this.state.content}</p>}
				</li>
			);
		});
	}

	render() {
		return (
			<div>
				<ul>
                    {this.handleDisplay()}
                </ul>
			</div>
		);
	}
}

export default Accordion;
