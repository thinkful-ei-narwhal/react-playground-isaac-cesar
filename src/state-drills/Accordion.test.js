import React from 'react';
import renderer from 'react-test-renderer'
import Accordion from'./Accordion'

describe('accordion',()=>{

    it('renders empty ul when section props null',() =>{
        expect(
            renderer.create(<Accordion />).toJSON()
            ).toMatchSnapshot()
    })
})
