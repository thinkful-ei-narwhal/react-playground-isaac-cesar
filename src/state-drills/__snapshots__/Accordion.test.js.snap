// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accordion renders empty  when section props null 1`] = `
<div>
  <ul />
</div>
`;

exports[`accordion renders empty li when section props null 1`] = `
<div>
  <ul />
</div>
`;

exports[`accordion renders empty ul when section props null 1`] = `
<div>
  <ul />
</div>
`;
